src/cart/crt_context.c

```diff
@@ -295,6 +299,9 @@ crt_rpc_complete(struct crt_rpc_priv *rpc_priv, int rc)
 	else
 		rpc_priv->crp_state = RPC_STATE_COMPLETED;
 
+	if (rc == -DER_TIMEDOUT)
+		d_tm_increment_counter(&rpc_timeouts, "RPC", "timeouts", NULL);
```

Comment:
This function is called on the client side too. Could you please elaborate on what happens there?
On the server side, i think we said that this should be reported per target.

Reply:
In the current implementation, when the client side calls d_tm_increment_counter, the function validates if the T&M is initialized for this instance or not, determines that it is not initialized, and returns -DER_UNINIT.  The caller is not processing these return values, so there is nothing logged.  The return codes are available for a developer to use as they are developing their code.  My expectation is that the examination of the return value would not make it into the production code once the developer verifies that everything is working as it should.

Reply:
Hmm. I don't think we need to do it now, but it seems like it would be better to use `#ifdef` to control when this is included in the compile. Then the client doesn't even see it. Also, I'm not sure that I agree with skipping the checking of return codes. It's true that a failure in the telemetry library shouldn't necessarily be fatal, but I'm concerned that we might be missing some important clues if/when things go wrong. I'll defer to @johannlombardi and other core DAOS developers on this point though.

Reply:
If the DAOS core devs agree to using #ifdef to exclude those calls from the client, that would be a great solution.  If we do get to use #ifdef to exclude them, then it makes it easy to add the checks for errors without generating noise elsewhere.